### ISSUE TITLE
transports

### ISSUE THREAD

**Issue by larshp** [EduardoCopat/code-review-with-abapgit#2]
btw, I have a planned feature: https://github.com/larshp/abapGit/issues/263

which would need to detect when something like https://github.com/EduardoCopat/code-review-with-abapgit/issues/1 happens

**Comment by eduardocopat** [comment]
@larshp I was thinking of automating this. 

Do you think it would be possible to call abapGit dynamically and generate a .zip of the transport in the background?

Looking at lcl_transport and lcl_zip, I see some possibilities.

The next step would be to send to the zip to a Continuous Integration pipeline

**Comment by larshp** [comment]
yes, it is possible to automate, take a look at https://gist.github.com/larshp/9137612a0b758dfc43c6d4e9200a1bc7 and https://gist.github.com/larshp/5ab1c0325b1f3068eed77b6220311108 on how to hook into the abapGit functions

How about having it automatically create branches for each transport like https://github.com/larshp/abapGit/issues/263, is it not possible for you to use online projects?

Note that the zip of the transport only contains the objects changed, which might be okay for doing reviews. However linting/unit testing requires all the other objects used in the code

**Comment by eduardocopat** [comment]
@larshp Thanks as always!

I thought if this couldn't be a feature on abapGit "Transport to branch".

But, maybe this is, "scope creep" into abapGit, could be something external.

As a first approach, I'd worry only about the code review, later the linting/testing.

**Comment by larshp** [comment]
I also need the "Transport to branch" feature for some stuff I'm planning, and would like it to happen automatically for online repos

**Comment by eduardocopat** [comment]
I'll give it a try this week then, I've been busy with other stuff and could contribute to abapGit lately... Although I always read issues and pull requests

**Comment by giovannidegani** [comment]
Not sure I understand the why need of the ZIP file. I was thinking more like dynamically create the branch in  Github ( Or TFS in my case ) and commit the stuff to it directly, then and branch hooks could trigger CI.

**Comment by larshp** [comment]
@tiefox yeah, me too. But I think Eduardo just want to test it with zips first to see how it works out

**Comment by eduardocopat** [comment]
@tiefox Yeah, that is also possible. I used the transport to zip feature because I was envisioning offline projects.

**Comment by eduardocopat** [comment]
Anyway, after https://github.com/larshp/abapGit/issues/263 is done, I'll update these slides also.
